packages/api/src/useAPI.ts

```diff
@@ -12,6 +12,14 @@ import APIContext from './APIContext';
 
 const useAPI = () => {
     const api = useContext(APIContext);
+    const readyState = api?.get()?.connection?.readyState;
+    const [isReady, setIsReady] = useState<boolean>(readyState === WebSocket.OPEN);
+
+    useEffect(() => {
+        if ((readyState === WebSocket.OPEN) !== isReady) {
+            setIsReady(readyState === WebSocket.OPEN);
+        }
+    }, [isReady, readyState]);
```

Comment:
I dunno. I think it might need something like this.

```suggestion
    useEffect(() => {
        if (!connection) return;

        const syncWithReadyState = () => {
            const readyState = api?.get()?.connection?.readyState;

            if (readyState === WebSocket.OPEN) {
                setIsReady(true);
                return;
            }

            setIsReady(false);
        };

        connection.addEventListener('open', syncWithReadyState);
        connection.addEventListener('close', syncWithReadyState);

        return () => {
            connection.removeEventListener('open', syncWithReadyState);
            connection.removeEventListener('close', syncWithReadyState);
        };
    }, [connection]);
```

Reply:
@markwylde-deriv as I said before I tried to use eventListener but it doesn't work. I used your code and I faced logout issue, with my implementation, no.

<img width="1537" alt="image" src="https://github.com/binary-com/deriv-app/assets/120570511/3e9b96ba-bbe3-470f-b32a-6342d9df0ba0">
<img width="1062" alt="image" src="https://github.com/binary-com/deriv-app/assets/120570511/dd98410a-1ab6-497e-9f09-4ab75cc0458e">

Reply:
Without events, nothing will trigger when the websocket connection state changes.

Reply:
@markwylde-deriv when I call it inside of the useEffect - the same result. If I call it outside of the useEffect (and create this function there) - too many renders...

I fully understand what you mean about trigger, and I don't fully understand why it works too, but it looks like `useQuery` creates and destroys every time when it calls. It means every `useQuery` calls own `useAPI` and every `useAPI` has own `isReady`. After calling `useQuery` in the hook it destroys, maybe it's the cause of why we have fresh `readyState` there.

It works even this way
<img width="646" alt="image" src="https://github.com/binary-com/deriv-app/assets/120570511/dc058e5d-9d6a-48e7-80e9-8106e31ce67a">

So I'll leave it like this :)